### PR TITLE
Feat/improve unity building

### PR DESCRIPTION
**Pull request by Ahmadre**
## Issues

https://github.com/juicycleff/flutter-unity-view-widget/issues/643

## Description

- This PR fixes latest Android Build and Run Performance (see https://github.com/juicycleff/flutter-unity-view-widget/issues/643 -> Thanks to: @timbotimbo)
- Adds iOS Debug and Release Builds
- Adds Android Debug and Release Builds
- Fixes iOS XCode Settings for Unity < 2020
- Updates actual compatibility to Unity 2022.1.7f1

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [x] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [x] 🧹 Code refactor
- [x] ✅ Build configuration change

### CODE REVIEW

**Review by juicycleff — APPROVED** [review]

**Review by dawiddszewczyk — APPROVED** [review]
For me nice, remember to update docs about this changes (related / debug version) :D

File: example/unity/DemoApp/Assets/FlutterUnityIntegration/Editor/Build.cs [hunk 1]
```diff
@@ -189,7 +203,17 @@ private static void DoBuildAndroid(String buildPath, bool isPlugin)
             playerOptions.scenes = GetEnabledScenes();
             playerOptions.target = BuildTarget.Android;
             playerOptions.locationPathName = APKPath;
-            playerOptions.options = BuildOptions.AllowDebugging;
+            if (!isReleaseBuild)
+            {
+                playerOptions.options = BuildOptions.AllowDebugging;
+            }
+            #if UNITY_2022_1_OR_NEWER
+				PlayerSettings.SetIl2CppCompilerConfiguration(BuildTargetGroup.Android, Il2CppCompilerConfiguration.Debug);
```
Comment (timbotimbo) [inline example/unity/DemoApp/Assets/FlutterUnityIntegration/Editor/Build.cs:211]: @Ahmadre 
Shouldn't these player settings be affected by `isReleaseBuild` too?

Comment (Ahmadre) [reply example/unity/DemoApp/Assets/FlutterUnityIntegration/Editor/Build.cs:211]: Thank you @timbotimbo ! You're totally right. Will add a check here.

Comment (Ahmadre) [reply example/unity/DemoApp/Assets/FlutterUnityIntegration/Editor/Build.cs]: @timbotimbo Done. Can we do that also for iOS Target? Or does that lead to errors?

Comment (timbotimbo) [reply example/unity/DemoApp/Assets/FlutterUnityIntegration/Editor/Build.cs]: I missed the other setting on the next line in my code quote. `Il2CppCodeGeneration.OptimizeSize` might worsen performance in release builds too.

I haven't tested these settings on iOS yet. In theory they shouldn't break anything, just make building a little faster and reduce runtime performance a little. But its always a good idea to test it first.

Comment (Ahmadre) [reply example/unity/DemoApp/Assets/FlutterUnityIntegration/Editor/Build.cs]: Alright. Then I will test it now with a production app on my side. 

Actually it's only iOS. So If it doesn't change significally I will leave iOS out.

Comment (Ahmadre) [reply example/unity/DemoApp/Assets/FlutterUnityIntegration/Editor/Build.cs]: Ok, I tested it and it's very fast on my iOS device! I really feel the "Release" Build :D

**Pull request merged**